gef.py

```diff
@@ -9250,12 +9255,67 @@ def do_invoke(self, argv: List[str]) -> None:
             else:
                 color = self["function_resolved"]
 
-            line = f"[{hex(address_val)}] "
-            line += Color.colorify(f"{name} {RIGHT_ARROW} {hex(got_address)}", color)
+            line = self.build_line(name, color, address_val, got_address)
             gef_print(line)
         return
 
 
+@register
+class GotAuditCommand(GotCommand):
+    """Display current status of the got inside the process with paths providing functions."""
+
+    _cmdline_ = "got-audit"
+    _syntax_ = f"{_cmdline_} [FUNCTION_NAME ...] "
+    _example_ = "got-audit read printf exit"
+    _symbols_ = collections.defaultdict(list)
+    _paths_ = collections.defaultdict(list)
+    _expected_dups_ = ['__cxa_finalize']
+
+    def get_symbols_from_path(self, elf_file):
+        nm = gef.session.constants["nm"]
+        # retrieve symbols using nm
+        lines = gef_execute_external([nm, "-D", elf_file], as_list=True)
+        for line in lines:
+            words = line.split(' ')
+            # Record the symbol if it is in the text section or
+            # an indirect function or weak symbol
+            if words[-2] in ('T', 'i', 'I', 'v', 'V', 'w', 'W'):
+                sym = words[-1].split('@')[0]
+                if elf_file not in self._symbols_[sym]:
+                    self._symbols_[sym].append(elf_file)
+                self._paths_[elf_file].append(sym)
+    
+    @only_if_gdb_running
+    def do_invoke(self, argv: List[str]) -> None:
+        # Build a list of the symbols provided by each path, and
+        # a list of paths that provide each symbol.
+        for section in gef.memory.maps:
+            if (hasattr(section, "path") and section.path not in self._paths_
```

Comment:
`Section` always has the `path` attribute, so this check will always return true. This part should be necessary